code/modules/mob/say.dm

```diff
@@ -68,14 +84,14 @@
 //returns the message mode string or null for no message mode.
 //standard mode is the mode returned for the special ';' radio code.
 /mob/proc/parse_message_mode(var/message, var/standard_mode="headset")
-	if(length(message) >= 1 && copytext_char(message,1,2) == get_prefix_key(/decl/prefix/radio_main_channel))
+	if(length(message) <= 0)
+		return null
+	var/initial_char = copytext_char(message,1,2)
+	if(initial_char == get_common_radio_prefix())
 		return standard_mode
-
-	if(length(message) >= 2)
-		var/channel_prefix = copytext_char(message, 1 ,3)
-		return department_radio_keys[channel_prefix]
-
-	return null
+	if(initial_char == get_department_radio_prefix() && length(message) >= 2)
+		var/channel_prefix = copytext(message, 2, 3)
```

Comment:
Should this use `copytext_char()`?

Reply:
If it does, `length` should use the char variant as well.